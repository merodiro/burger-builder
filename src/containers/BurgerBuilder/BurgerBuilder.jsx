import React, { Component, Fragment } from 'react'

import Burger from '../../components/Burger/Burger'
import BuildControls from '../../components/Burger/BuildControls/BuildControls'

const INGREDIENT_PRICES = {
  salad: 0.5,
  cheese: 0.4,
  meat: 1.3,
  bacon: 0.7
}

export default class BurgerBuilder extends Component {

  state = {
    ingredients: {
      salad: 0,
      bacon: 0,
      cheese: 0,
      meat: 0
    },
    totalPrice: 4
  }

  addIngredientHandled = (type) => {
    this.setState((oldState, props) => {
      return {
        totalPrice: oldState.totalPrice + INGREDIENT_PRICES[type],
        ingredients: {
          ...oldState.ingredients,
          [type]: oldState.ingredients[type] + 1
        }
      }
    })
  }

  removeIngredientHandled = (type) => {
    if (this.state.ingredients[type] <= 0) {
      return
    }
    this.setState((oldState, props) => ({
      totalPrice: oldState.totalPrice - INGREDIENT_PRICES[type],
      ingredients: {
        ...oldState.ingredients,
        [type]: oldState.ingredients[type] - 1
      }
    }))
  }


  render() {
    let { ingredients } = this.state
    const disabledInfo = {}
    Object.keys(ingredients).forEach(key => {
      disabledInfo[key] = ingredients[key] <= 0
    });
    return (
      <Fragment>
        <Burger ingredients={this.state.ingredients} />
        <BuildControls
          price={this.state.totalPrice}
          disabled={disabledInfo}
          ingredientAdded={this.addIngredientHandled}
          ingredientRemoved={this.removeIngredientHandled}
        />
      </Fragment>
    )
  }
}
