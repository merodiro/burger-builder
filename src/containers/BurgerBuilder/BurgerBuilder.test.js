import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import BurgerBuilder from './BurgerBuilder'

describe('<BurgerBuilder>', () => {
  it('renders the same output', () => {
    const component = shallow(<BurgerBuilder />)

    expect(component).toMatchSnapshot()
  })

  it('renders with ingredents', () => {
    const component = shallow(<BurgerBuilder />)
    component.setState({
      ingredients: {
        salad: 1,
        bacon: 1,
        cheese: 2,
        meat: 1,
      },
    })
    expect(component).toMatchSnapshot()
  })
})
