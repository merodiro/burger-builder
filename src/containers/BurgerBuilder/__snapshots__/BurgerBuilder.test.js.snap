// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerBuilder> renders the same output 1`] = `
<React.Fragment>
  <Modal
    closed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "bacon": 0,
          "cheese": 0,
          "meat": 0,
          "salad": 0,
        }
      }
      price={4}
      purchaseCancel={[Function]}
      purchaseContinue={[Function]}
    />
  </Modal>
  <_default
    ingredients={
      Object {
        "bacon": 0,
        "cheese": 0,
        "meat": 0,
        "salad": 0,
      }
    }
  />
  <BuildControls
    disabled={
      Object {
        "bacon": true,
        "cheese": true,
        "meat": true,
        "salad": true,
      }
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={4}
    purchasable={false}
  />
</React.Fragment>
`;

exports[`<BurgerBuilder> renders with ingredents 1`] = `
<React.Fragment>
  <Modal
    closed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "bacon": 1,
          "cheese": 2,
          "meat": 1,
          "salad": 1,
        }
      }
      price={7.3}
      purchaseCancel={[Function]}
      purchaseContinue={[Function]}
    />
  </Modal>
  <_default
    ingredients={
      Object {
        "bacon": 1,
        "cheese": 2,
        "meat": 1,
        "salad": 1,
      }
    }
  />
  <BuildControls
    disabled={
      Object {
        "bacon": false,
        "cheese": false,
        "meat": false,
        "salad": false,
      }
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={7.3}
    purchasable={true}
  />
</React.Fragment>
`;
